### PR TITLE
Auth provider added

### PR DESCRIPTION
**Pull request by animeshgarai09**
Files to check: -

[src/store/providers/auth-provider.js](https://github.com/animeshgarai09/Felix-Store-React/pull/6/files#diff-a9bc401ac60662737dd3de8488b5cfbc2f354937b62cf01ffc896b75b3073b0d)
[src/pages/authentication/authentication.js](https://github.com/animeshgarai09/Felix-Store-React/pull/6/files#diff-e8e1b89f4668f1edbddffcb279a3c7cc68e58d8ff577e2356e4a02c8d28f0f55)
[src/components/header/header.js](https://github.com/animeshgarai09/Felix-Store-React/pull/6/files#diff-482aadf97cfd2f2ae994bc8f3c687f4e1178f3b3ecf79d2fd6e3196377c1580b)
[src/pages/authentication/sub-components/signin.js](https://github.com/animeshgarai09/Felix-Store-React/pull/6/files#diff-9722342b35a2e13e099cad4d2c4bdc89a1aeb9d7bb2e247031df10a8d28a46a4)
[src/pages/authentication/sub-components/signup.js](https://github.com/animeshgarai09/Felix-Store-React/pull/6/files#diff-1cd880da7c2ab523eac6378fb035fab6eda4e236fd4c2dc0a2d449d42825c6bd)

### CODE REVIEW

**Review by anik31 — APPROVED** [review]
Nice work, code is modular & readable 💯

File: src/components/header/header.js [hunk 1]
```diff
@@ -68,7 +71,11 @@ const Header = () => {
                                 </ListItem>
                                 <ListItem>
                                     <Link to="/signin" className={styles.link}>
-                                        <IconButton icon={<RiUser6Fill />} className={styles.icon} />
+                                        {userState._id !== ""
+                                            ? <Avatar size="sm" name={userState.name} className={styles.avatar} />
+                                            : <IconButton icon={<RiUser6Fill />} className={styles.icon} />
+                                        }
+                                        {/* <IconButton icon={<RiUser6Fill />} className={styles.icon} /> */}
```
Comment (zoef-yt) [inline src/components/header/header.js:78]: Remove commented code.

**Review by zoef-yt — APPROVED** [review]
The code looks good. The abstraction of the code is really great. The UI looks great. The only thing is that the preview on Vercel is not working do check it out.  Good to merge.

File: src/components/header/header.js (hunk 1 above)
Comment (animeshgarai09) [reply src/components/header/header.js:78]: Thanks, will remove it. [preview](https://felix-store-git-auth-provider-animeshgarai09.vercel.app/) this link is working.

**Pull request merged**